netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -527,208 +527,6 @@ static void logSslEngineDetails(Level level, ChannelHandlerContext ctx, String m
     log.log(level, builder.toString(), t);
   }
 
-  /**
-   * Buffers all writes until either {@link #writeBufferedAndRemove(ChannelHandlerContext)} or
-   * {@link #fail(ChannelHandlerContext, Throwable)} is called. This handler allows us to
-   * write to a {@link io.netty.channel.Channel} before we are allowed to write to it officially
-   * i.e.  before it's active or the TLS Handshake is complete.
-   */
-  public abstract static class AbstractBufferingHandler extends ChannelDuplexHandler {
```

Comment:
Ideally this removal would have been kept as a separate commit/PR to allow backporting the httpProxy work. But I guess there's no point in splitting it up now.

Reply:
good point, i'll remember to split!